contrib/vcl/source/BUILD

```diff
@@ -28,37 +29,63 @@ cc_library(
     visibility = ["//visibility:public"],
 )
 
-genrule(
+envoy_cmake(
     name = "build",
-    srcs = [
-        "@com_github_fdio_vpp_vcl//:all",
+    build_data = [requirement("ply")],
+    cache_entries = {
+        "CMAKE_BUILD_TYPE": "Release",
+        "VPP_API_TEST_BUILTIN": "OFF",
+        "BUILD_SHARED_LIBS": "OFF",
+        "CMAKE_ENABLE_TESTING": "OFF",
+        "CMAKE_ENABLE_EXPORTS": "OFF",
+    },
+    # build_args = ["--", "-C", ".", "vppcom"],
+    copts = ["-Wno-unused-variable"],
+    default_cache_entries = {},
+    env = {
+        "PLYPATHS": "$(locations %s)" % requirement("ply"),
+    },
+    lib_source = "@com_github_fdio_vpp_vcl//:all",
+    linkopts = ["-Wno-unused-variable"],
+    out_static_libs = [
+        "libvppcom.a",
+        "libvppinfra.a",
+        "libsvm.a",
+        "libvlibmemoryclient.a",
+    ],
+    postfix_script = """
+        mkdir -p $INSTALLDIR/lib/external $INSTALLDIR/include/external \
+        && find . -name "*.a" | xargs -I{} cp -a {} $INSTALLDIR/lib/ \
+        && find . -name "*.h" | xargs -I{} cp -a {} $INSTALLDIR/include
+    """,
+    tags = [
+        "cpu:16",
+        "skip_on_windows",
     ],
+    # tags = ["no-remote", "skip_on_windows"],
+    targets = [
+        "",
+        "install",
+    ],
+    working_directory = "src",
+)
+
+genrule(
```

Comment:
Could we maybe leverage something like [this](https://github.com/envoyproxy/envoy/pull/18149/commits/7f589cc7eff951d66f626589270d303e71707ded#diff-a6a4eb9af2d1f92545242b7adfe0eda8d96cca4f8d6f2b55902846da59e8502b) first approach to building vppcom?

Reply:
not sure which bit of that approach you mean, but that is the issue

ideally we do this with rules_foreign_cc - doing it with a genrule **requires** getting a hermetic cmake toolchain, which i think is not trivial with foreign_cc (not sure why the documented example doesnt seem to work, and it was for `make` - there was no equivalent setup for cmake anyway)

Reply:
for ref the `make` example

https://github.com/bazelbuild/rules_foreign_cc/blob/7fa1a1259bbc4fbceb4dc3a23ed520907b5d8d1d/examples/current_toolchains/BUILD.bazel#L1-L13

Reply:
> getting a hermetic cmake toolchain, which i think is not trivial with foreign_cc 

this should be easy - ive certainly found it so with other toolchains that ive worked with

possibly we could do something similar to how we fish out the zstd bin here https://github.com/envoyproxy/envoy/blob/main/tools/zstd/zstd.bzl - altho its a little different as cmake/ninja are provided as a toolchain rather than just a tool

tbh tho its just fighting the system - the better, easier to maintain way would be to get the cmake <> foreign_cc setup correct i think